frontend/src/routes/password.change.tsx

```diff
@@ -0,0 +1,293 @@
+                  onBlur={() => newPasswordAgainRef.current!.reportValidity()}
+                />
+
+                {/* TODO Show a password bar. https://github.com/matrix-org/matrix-authentication-service/issues/2854 */}
+
+                <Form.ErrorMessage match="valueMissing">
+                  {t("frontend.errors.field_required")}
+                </Form.ErrorMessage>
+              </Form.Field>
+
+              <Form.Field name="new_password_again">
+                {/*
+                TODO This field has validation defects,
+                some caused by Radix-UI upstream bugs.
+                https://github.com/matrix-org/matrix-authentication-service/issues/2855
+              */}
+                <Form.Label>
+                  {t("frontend.password_change.new_password_again_label")}
+                </Form.Label>
+
+                <Form.PasswordControl
+                  required
+                  ref={newPasswordAgainRef}
+                  autoComplete="new-password"
+                />
+
+                <Form.ErrorMessage match="valueMissing">
+                  {t("frontend.errors.field_required")}
+                </Form.ErrorMessage>
+
+                <Form.ErrorMessage
+                  match={(v, form) => v !== form.get("new_password")}
+                >
+                  {t("frontend.password_change.passwords_no_match")}
+                </Form.ErrorMessage>
+
+                <Form.HelpMessage match="valid">
+                  {/* TODO Use SuccessMessage once ready. https://github.com/matrix-org/matrix-authentication-service/issues/2856 */}
+                  <IconCheckCircleSolid />
+                  {t("frontend.password_change.passwords_match")}
+                </Form.HelpMessage>
+              </Form.Field>
+            </>
+          )}
+
+          {unhandleableError && (
+            <Alert
+              type="critical"
+              title={t("frontend.password_change.failure.title")}
+            >
+              {t("frontend.password_change.failure.description")}
+            </Alert>
+          )}
+          {!success && (
+            <>
+              <Form.Submit kind="primary" disabled={result.fetching}>
+                {!!result.fetching && <LoadingSpinner inline />}
+                {t("action.save")}
+              </Form.Submit>
+            </>
+          )}
+          {success && (
+            <Alert
+              type="success"
+              title={t("frontend.password_change.success.title")}
+            >
+              {t("frontend.password_change.success.description")}
+            </Alert>
+          )}
+          {handleableError && (
+            <Alert
+              type="critical"
+              title={t("frontend.password_change.failure.title")}
+            >
+              {errorMsg}
+            </Alert>
+          )}
+
+          <ButtonLink to="/" kind="tertiary">
+            {(result.data && t("action.back")) || t("action.cancel")}
+          </ButtonLink>
```

Comment:
Making a dedicated success page would avoid having this kind of logic within the component